net/http/src/THttpServer.cxx

```diff
@@ -924,18 +828,19 @@ void THttpServer::ProcessRequest(THttpCallArg *arg)
          // if accepted, reply with connection id, which must be used in the following communications
          arg->SetMethod("WS_CONNECT");
 
-         if (handler && handler->HandleWS(arg)) {
-            arg->SetMethod("WS_READY");
-
-            TLongPollEngine *handle = new TLongPollEngine("longpoll", arg->fPathName.Data());
+         THttpLongPollEngine *handle = new THttpLongPollEngine("longpoll", arg->fPathName.Data());
+         arg->SetWSId(handle->GetId());
 
-            arg->SetWSId(handle->GetId());
+         if (handler->HandleWS(arg)) {
+            arg->SetMethod("WS_READY");
             arg->SetWSHandle(handle);
 
             if (handler->HandleWS(arg)) {
                arg->SetContent(TString::Format("%u", arg->GetWSId()));
                arg->SetContentType("text/plain");
```

Comment:
Also here I'd *much* prefer to have an enum, say `EContentType::kTextPlain`.

Reply:
It is again field in HTTP header. And in special usecases one should have possibility to set it directly. I rather prefer `arg->SetTextContentType()` plus few others.

Reply:
But you are creating a C++ interface around this. In the end it's a string, sure - but the C++ interface should be robust. `"test/plein"` is not :-)

Reply:
> "test/plein" is not :-)

But finally it can be valid value - it is just configuration of your browser or that ever HTTP client user will use